skycatalogs/objects/gaia_object.py

```diff
@@ -126,6 +135,95 @@ def set_use_lut(self, use_lut):
         self.use_lut = use_lut
 
 
+def _read_fits(htm_id, gaia_config, out_dict, region=None, silent=True):
+    '''
+    Read data for columns in keys from fits file belonging to htm_id.
+    Append to arrays in out_dict.  If region is not None, filter out
+    entries not in region before appending.
+
+    Parameters
+    ----------
+    htm_id         int
+    gaia_config    dict     gaia_star portion of skyCatalg config
+    out_dict       dict     out_dict.keys() are the columns to store
+    region         lsst.sphgeom.Region or None
+    silent         bool     if False print warning if file not found.
+                            Default is False.
+    '''
+    f_dir = gaia_config['data_dir']
+    f_name = gaia_config['basename_template'].replace('(?P<htm>\\d+)',
+                                                      f'{htm_id}')
+    f_path = os.path.join(f_dir, f_name)
+    try:
+        f = open(f_path)
+        f.close()
+    except FileNotFoundError:
+        if not silent:
+            print(f'No file for requested htm id {htm_id}')
+        return
+
+    tbl = afwtable.SimpleCatalog.readFits(f_path)
+    if region is None:
+        for k in out_dict.keys():
+            out_dict[k] += list(tbl.get(k))
+        return
+
+    # Otherwise start with ra, dec and create mask
+    ra_full = tbl.get('coord_ra')
+    dec_full = tbl.get('coord_dec')
```

Comment:
I'm pretty sure `coord_ra` and `coord_dec` are in radians in the refcat files-- see the conversion in `GaiaObject.__init__`.   If so, these need to be converted to degrees when computing the masks below.